lib/rule-tester/rule-tester.js

```diff
@@ -499,6 +500,9 @@ class RuleTester {
             linter = this.linter,
             ruleId = `rule-to-test/${ruleName}`;
 
+        const seenValidTestCases = new Set();
+        const seenInvalidTestCases = new Set();
```

Comment:
[Question] Out of curiosity: is there a hard need for _two_ `Set`s? Is there realistically ever a situation where an `invalid` case has all the same checked properties as a `valid` case?

Reply:
I think you're right we could use one set if we wanted. However, I think it's cleaner to separate them, and theoretically, it could have faster performance to separate them (ideally, set lookup should be O(1) but could be worse in the case of hash collisions from storing a large number of items).